src/Common/tests/System/RealFormatterTestsBase.netcoreapp.cs

```diff
@@ -14,29 +14,118 @@ public abstract class RealFormatterTestsBase
+        public void TestFormatterDouble_Custom(string format, double value, string expectedResult) => TestFormatterDouble_Standard(value, format, expectedResult);
+
         [Theory]
         [InlineData(double.Epsilon, "¤0.00")]
         [InlineData(double.MaxValue, "¤179,769,313,486,231,570,814,527,423,731,704,356,798,070,567,525,844,996,598,917,476,803,157,260,780,028,538,760,589,558,632,766,878,171,540,458,953,514,382,464,234,321,326,889,464,182,768,467,546,703,537,516,986,049,910,576,551,282,076,245,490,090,389,328,944,075,868,508,455,133,942,304,583,236,903,222,948,165,808,559,332,123,348,274,797,826,204,144,723,168,738,177,180,919,299,881,250,404,026,184,124,858,368.00")]
         [InlineData(Math.E, "¤2.72")]
         [InlineData(Math.PI, "¤3.14")]
         [InlineData(0.0, "¤0.00")]
         [InlineData(0.0046, "¤0.00")]
+        [InlineData(0.005, "¤0.01")]
         [InlineData(0.125, "¤0.12")]
+        [InlineData(0.5, "¤0.50")]
+        [InlineData(0.51, "¤0.51")]
+        [InlineData(0.56789, "¤0.57")]
         [InlineData(0.84551240822557006, "¤0.85")]
         [InlineData(1.0, "¤1.00")]
+        [InlineData(505.0, "¤505.00")]
+        [InlineData(506.0, "¤506.00")]
+        [InlineData(545.0, "¤545.00")]
+        [InlineData(545.1, "¤545.10")]
+        [InlineData(555.0, "¤555.00")]
+        [InlineData(65747.125, "¤65,747.12")]
         [InlineData(1844674407370955.25, "¤1,844,674,407,370,955.25")]
         public void TestFormatterDouble_C(double value, string expectedResult) => TestFormatterDouble_Standard(value, "C", expectedResult);
 
+        [Theory]
+        [InlineData(double.Epsilon, "¤0")]
+        [InlineData(double.MaxValue, "¤179,769,313,486,231,570,814,527,423,731,704,356,798,070,567,525,844,996,598,917,476,803,157,260,780,028,538,760,589,558,632,766,878,171,540,458,953,514,382,464,234,321,326,889,464,182,768,467,546,703,537,516,986,049,910,576,551,282,076,245,490,090,389,328,944,075,868,508,455,133,942,304,583,236,903,222,948,165,808,559,332,123,348,274,797,826,204,144,723,168,738,177,180,919,299,881,250,404,026,184,124,858,368")]
+        [InlineData(Math.E, "¤3")]
+        [InlineData(Math.PI, "¤3")]
+        [InlineData(0.0, "¤0")]
+        [InlineData(0.0046, "¤0")]
+        [InlineData(0.005, "¤0")]
+        [InlineData(0.125, "¤0")]
+        [InlineData(0.5, "¤0")]
+        [InlineData(0.51, "¤1")]
+        [InlineData(0.56789, "¤1")]
+        [InlineData(0.84551240822557006, "¤1")]
+        [InlineData(1.0, "¤1")]
+        [InlineData(505.0, "¤505")]
+        [InlineData(506.0, "¤506")]
+        [InlineData(545.0, "¤545")]
+        [InlineData(545.1, "¤545")]
+        [InlineData(555.0, "¤555")]
+        [InlineData(65747.125, "¤65,747")]
+        [InlineData(1844674407370955.25, "¤1,844,674,407,370,955")]
+        public void TestFormatterDouble_C0(double value, string expectedResult) => TestFormatterDouble_Standard(value, "C0", expectedResult);
```

Comment:
Also added explicit format groups for precisions of `0` and precisions of `2`.

Both tend to have interesting characteristics for the given test inputs that I listed as being added above.